modules/nf-commons/src/main/nextflow/plugin/PluginUpdater.groovy

```diff
@@ -316,14 +313,25 @@ class PluginUpdater extends UpdateManager {
     private boolean load0(String id, String version) {
         assert id, "Missing plugin Id"
 
-        if( version == null )
-            version = getLastPluginRelease(id)?.version
-        if( !version )
-            throw new IllegalStateException("Cannot find latest version of $id plugin")
+        def pluginPath = null
+        if( SysEnv.get('NXF_OFFLINE')=='true' ) {
+            if( version == null )
+                version = getLastPluginReleaseOffline(id)
+            if( !version )
+                throw new IllegalStateException("Cannot find local version of $id plugin (offline mode)")
 
-        def pluginPath = pluginsStore.resolve("$id-$version")
-        if( !FilesEx.exists(pluginPath) ) {
-            pluginPath = safeDownload(id, version)
+            log.debug "Update disabled in offline mode -- using local version $version of $id plugin"
+            pluginPath = pluginsStore.resolve("$id-$version")
+        }
+        else {
+            if( version == null )
+                version = getLastPluginRelease(id)?.version
+            if( !version )
+                throw new IllegalStateException("Cannot find latest version of $id plugin")
+
+            pluginPath = pluginsStore.resolve("$id-$version")
+            if( !FilesEx.exists(pluginPath) )
+                pluginPath = safeDownload(id, version)
```

Comment:
This change is too convoluted. Ideally it should something like 

```
        if( !version )
            version = offline ? getLastPluginReleaseOffline(id) : getLastPluginRelease(id)
```

Reply:
I would like to do that too, but the error messages are different and the online path also calls `safeDownload()`. You think it would be better to just check `offline` multiple times?

Reply:
but once the version is resolved (not null) the `safeDownload` just use the local installation if it can find it

Reply:
true 👍